lib/mongodb.js

```diff
@@ -768,45 +768,113 @@ function convertToMeters(distance, unit) {
   }
 }
 
-function buildNearFilter(query, near) {
-  var coordinates = {};
-  if (typeof near.near === 'string') {
-    var s = near.near.split(',');
-    coordinates.lng = Number(s[0]);
-    coordinates.lat = Number(s[1]);
-  } else {
-    coordinates = near.near;
+function buildNearFilter(query, params) {
+  if (!Array.isArray(params)) {
+    params = [params];
   }
 
-  query.where[near.key] = {
-    near: {
-      $geometry: {
-        coordinates: [coordinates.lng, coordinates.lat],
-        type: 'Point',
+  params.forEach(function(near) {
+    var coordinates = {};
+
+    if (typeof near.near === 'string') {
+      var s = near.near.split(',');
+      coordinates.lng = parseFloat(s[0]);
+      coordinates.lat = parseFloat(s[1]);
+    } else {
+      coordinates = near.near;
+    }
+
+    var props = ['maxDistance', 'minDistance'];
+    // use mongodb default unit 'meters' rather than 'miles'
+    var unit = near.unit || 'meters';
+
+    var queryValue = {
+      near: {
+        $geometry: {
+          coordinates: [coordinates.lng, coordinates.lat],
+          type: 'Point',
+        },
       },
-    },
-  };
+    };
+
+    props.forEach(function(p) {
+      if (near[p]) {
+        queryValue.near['$' + p] = convertToMeters(near[p], unit);
+      }
+    });
+
+    var property;
+
+    if (near.key) {
+      if (Array.isArray(near.key)) {
+        // the key is an array, we need to handle the depth properly
+        property = query.where;
+
+        // loop through every level of depth of the key, to reach the right depth
+        for (var i = 0; i < near.key.length; i++) {
+          var subKey = near.key[i];
+          // while we're not still at the end (ie, there's at least one more item to go through)
+          if (near.key.hasOwnProperty(i + 1)) {
+            // if the query doesn't still have the current key set (undefined)
+            if (!property.hasOwnProperty(subKey)) {
+              // if the next level of depth is an integer, we create an array, otherwise an empty object
```

Comment:
I was thinking more along the lines of a single comment explaining the block:

// if key is array, recursively search all items in array for subkeys